### DISCUSSION TITLE
[solved] replace abbreviation not working in Libre Office Write

### DISCUSSION THREAD

**Discussion by unlucky67** [autokey/autokey#555] [General]
Did post about this on reddit and asking here was recommended. At first I thought it was due to using space & enter as a trigger but have discovered this is happening in Libre office write
Here I don't need to use space &enter - just space triggers text replacement and it works as expected
Eg abbreviation is mac
replacement text is  
Macaroni Cheese
Broccoli 
In LO write (again just with space) I get
mac macaroni cheese
brocc

I thought the trigger keys were shift and enter  - this works too 
except I get 
maMacaroni Cheese
Broccol

Also in LO other abbrev text replacements I have set up with any key trigger have the same issue with not deleting the abbreviation and truncating text.

Any suggestions? I really would like to use some of these text replacements in LO, also would prefer if I could have them not triggering with just the space bar - unless there is a way of turning some on and off when required.
Thanks 

![Screenshot autokey 2](https://user-images.githubusercontent.com/85312182/120704114-17f1ac80-c4ae-11eb-92f2-6e5d28e5c0c7.png)
![Autokey main screen](https://user-images.githubusercontent.com/85312182/120704024-f85a8400-c4ad-11eb-8aae-1dacd6c261e7.png)

**Comment by josephj11** [comment]
This sort of thing happens all the time. AutoKey types at machine speed and many applications can't keep up and lose characters. LibreOffice and the Firefox address bar have this problem.

TL;DR: Always use the Paste using clipboard (Ctrl+V) option for phrases unless you specifically need something it won't do.

This will *finally* be the default option in release 0.96.

Please also take a look at [this](https://github.com/autokey/autokey/wiki/Emitting-Keyboard-Events) new wiki page. It will save you a lot of time in the future.

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.

**Reply by unlucky67** [reply]
Thank you - that works perfectly. Do still have the issue that  they are triggered by space only 
(Actually just thought - does 'space AND enter' really mean space OR enter?)
I'll check out the links and post as solved on reddit too. 
Finding help is made more difficult by the top hits from searches being for Autohotkey (which I used when I was using windows) - not sure how you could stop that happening.  I know they are two different things but still get caught out sometimes! 
Thanks again

**Reply by josephj11** [reply]
Yes, the trigger characters are a list, any one of which will work.
BTW, we don't usually change issue titles to mark as solved, we just close the issue ad since you opened it, you have permission to close it.

**Reply by unlucky67** [reply]
I'm sorry ...I really can't see where the option to close it is... 
Thank you again for all your help and if there is anything I can do to help (even though I am obviously a novice and not a developer -guess I could do testing or something just let me know

**Comment by josephj11** [comment]
Join us on our list and/or Gitter (links above) to participate in whatever is going on. We're currently testing a beta release of 0.96 with a lot of fixes and some great new features. It takes a bit of CLI knowledge to get it installed, but we can help. I haven't quite installed it myself yet, but I setup my system for it.

Apparently, there's no way to close a discussion like you can an issue. That's another reason to handle support issues on our list and Gitter.
